### PR TITLE
Remove git hook support in favour of documentation

### PR DESCRIPTION
**Pull request by maelle**
cf #251 #267 #234 #168 and cf #260 (because it edits the navbar)

This also solves #272 :grimacing: 

Oh and this _adds_ a GH actions workflow that will keep codemeta.json in sync with DESCRIPTION for this repo @cboettig so quite a big change.

### CODE REVIEW

**Comment by maelle** [comment]
@cboettig is the deprecation ok? And are the docs clear? Is the get started vignette too long, should "how to improve your codemetar" go into the workflow vignette?

@Bisaloo your feedback is welcome too!

I haven't been able to make articles disappear from the navbar (there's intro and advanced, but also articles!)

**Comment by maelle** [comment]
Reg precommit here's my PR for adding a hook for codemeta https://github.com/lorenzwalthert/precommit/pull/60

**Comment by maelle** [comment]
@cboettig can we merge this? Particular aspects
* is the deprecation of the argument ok?
* this adds both a GitHub actions workflow template _and_ the workflow to this repo. so now when DESCRIPTION is edited in a commit, codemeta.json will be updated by GH actions. See [log](https://github.com/ropensci/codemetar/commit/dca3f1b476fec8289a726dcf495e78596fd2e020/checks?check_suite_id=355105167) and the [resulting commit](https://github.com/ropensci/codemetar/pull/271/commits/10dccc6174cb188c79b5a98fc13b86da6f8c2e23)
* the intro vignette got longer.

File: man/rmdhunks/uptodate.Rmd [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      
+    * If that's your only pre-commit hook (i.e. you don't have one created by e.g. `usethis::use_readme_rmd()`), then you can create it using
+    
+```r
+script = readLines(system.file("templates", "description-codemetajson-pre-commit.sh", package = "codemetar"))
+usethis::use_git_hook("pre-commit",
+                     script = script)
+```
+
+* You could use GitHub actions. Refer to GitHub actions docs <https://github.com/features/actions>, and to the example workflow provided in this package (type `system.file("templates", "codemeta-github-actions.yml", package = "codemetar")`). In that case don't forget to pull ([`git pull --rebase`](https://stackoverflow.com/questions/18930527/difference-between-git-pull-and-git-pull-rebase/38139843#38139843)) from upstream before you commit new changes.
```
Comment (Bisaloo) [inline man/rmdhunks/uptodate.Rmd]:
```suggestion
* You could use GitHub actions. Refer to GitHub actions docs <https://github.com/features/actions>, and to the example workflow provided in this package (type `system.file("templates", "codemeta-github-actions.yml", package = "codemetar")`). If you take this route, you should try to remember to run `git pull` before making any new changes on your local project. However, if you forgot to pull and already committed new changes, fret not, you can use ([`git pull --rebase`](https://stackoverflow.com/questions/18930527/difference-between-git-pull-and-git-pull-rebase/38139843#38139843)) to rewind you local changes on top of the current upstream `HEAD`.
```

Something like this maybe? Feel free to reword it according to your taste. This is just to give a starting point.

Comment (maelle) [reply man/rmdhunks/uptodate.Rmd]: I love the use of fret not! And it reads well, thanks!

File: inst/templates/codemeta-github-actions.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+name: Render codemeta
+jobs:
+  render:
+    name: Render codemeta
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: r-lib/actions/setup-r@v1
+      - name: Install codemetar
+        run: Rscript -e 'remotes::install.packages("codemetar")'
```
Comment (Bisaloo) [inline inst/templates/codemeta-github-actions.yml]: I'm currently testing this. Am I missing something or did you forget a line to install `remotes`? It's present in this repo `main.yml`

Comment (Bisaloo) [reply inst/templates/codemeta-github-actions.yml]: or 
```suggestion
        run: Rscript -e 'remotes::install_cran("codemetar")'
```

I don't know how caching works for GitHub actions.

**Pull request merged**